operators/nvidia_video_codec/nv_video_decoder/nv_video_decoder.cpp

```diff
@@ -0,0 +1,314 @@
+  if (!maybe_entity) {
+    throw std::runtime_error("Failed to receive input entity");
+  }
+
+  // Get the tensor from the input message
+  auto tensor = maybe_entity.value().get<Tensor>("");
+  if (!tensor) {
+    throw std::runtime_error("Failed to get tensor from input message");
+  }
+
+  auto data_size = tensor->size();
+  auto data_ptr = tensor->data();
+
+  file_data_provider_->SetData(static_cast<uint8_t*>(data_ptr), data_size);
+
+  if (demuxer_ == nullptr) {
+    // Set the current context
+    CudaCheck(cuCtxPushCurrent(cu_context_));
+
+    try {
+      demuxer_ = std::make_unique<FFmpegDemuxer>(file_data_provider_.get());
+      decoder_ = std::make_unique<NvDecoder>(cu_context_,
+                                             true,
+                                             FFmpeg2NvCodecId(demuxer_->GetVideoCodec()),
+                                             true,
+                                             false,
+                                             nullptr,
+                                             nullptr,
+                                             false,
+                                             0,
+                                             0,
+                                             1000,
+                                             true);
+    } catch (const std::exception& e) {
+      HOLOSCAN_LOG_ERROR("Failed to initialize decoder: {}", e.what());
+      // Reset the demuxer and decoder for potential retry
+      demuxer_.reset();
+      decoder_.reset();
+      // Pop the context to avoid context stack issues
+      CudaCheck(cuCtxPopCurrent(nullptr));
+      throw std::runtime_error(
+          "Decoder initialization failed. Please check video format and codec support.");
+    }
+  }
+
+  auto allocator =
+      nvidia::gxf::Handle<nvidia::gxf::Allocator>::Create(context.context(), allocator_->gxf_cid());
+
+  auto output = nvidia::gxf::Entity::New(context.context());
+  if (!output) {
+    throw std::runtime_error("Failed to allocate message for output");
+  }
+
+  auto maybe_video_buffer = output.value().add<nvidia::gxf::VideoBuffer>();
+  if (!maybe_video_buffer) {
+    throw std::runtime_error("Failed to allocate video buffer");
+  }
+
+  auto video_buffer = maybe_video_buffer.value();
+  nvidia::gxf::VideoTypeTraits<nvidia::gxf::VideoFormat::GXF_VIDEO_FORMAT_NV12> video_type;
+  nvidia::gxf::VideoFormatSize<nvidia::gxf::VideoFormat::GXF_VIDEO_FORMAT_NV12> color_format;
+
+  int nFrame = 0;
+  uint8_t* pVideo = NULL;
+  int nVideoBytes = 0;
+  uint8_t* pFrame;
+  int nFrameReturned = 0;
+
+  do {
+    demuxer_->Demux(&pVideo, &nVideoBytes);
+
+    try {
+      nFrameReturned = decoder_->Decode(pVideo, nVideoBytes);
+    } catch (const std::exception& e) {
+      HOLOSCAN_LOG_ERROR("Decode failed: {}", e.what());
+      // Skip this frame and continue
+      continue;
+    }
+
+    if (nFrameReturned == 0) {
+      // emit_empty_frame(op_output, context);
+      continue;
+    }
```

Comment:
Spec == no emit if decode fails?

```suggestion
    if (nFrameReturned == 0) {
      continue;
    }
```